### PR TITLE
Fix spawning of absolute-triggered tasks.

### LINKED ISSUE #5002: Parentless tasks try to spawn again after already finishing
#### Describe the bug

(Back compat mode, possible follow-up to #4906)

A failed, parentless task that was suicided tries to spawn again.

#### Steps to reproduce the bug
Take the following `suite.rc` running in back-compat mode.
```ini
[scheduling]
    cycling mode = integer
    initial cycle point = 1
    [[dependencies]]
        [[[P1]]]
            graph = """
                foo:fail => recover
                foo | recover => bar

                foo => !recover
                recover => !foo
            """
[runtime]
    [[foo]]
        script = false
    [[bar, recover]]
        script = true
```

You get a message that a `foo` suicides after `recover` succeeds. However, a bit further down you get "foo already spawned in {1}"

```
CRITICAL - [2/foo failed job:01 flows:1] task proxy removed (suicide)
INFO - [3/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:37+01:00
INFO - [3/recover running job:01 flows:1] => succeeded
INFO - [3/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [3/foo failed job:01 flows:1] task proxy removed (suicide)
INFO - [4/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:37+01:00
INFO - [4/recover running job:01 flows:1] => succeeded
INFO - [4/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [4/foo failed job:01 flows:1] task proxy removed (suicide)
INFO - [5/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:37+01:00
INFO - [5/recover running job:01 flows:1] => succeeded
INFO - [5/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [5/foo failed job:01 flows:1] task proxy removed (suicide)
...
INFO - [1/recover running job:01 flows:1] (received)succeeded at 2022-07-21T17:18:38+01:00
INFO - [1/recover running job:01 flows:1] => succeeded
WARNING - Task 2/foo already spawned in {1}
WARNING - Task 3/foo already spawned in {1}
WARNING - Task 4/foo already spawned in {1}
WARNING - Task 5/foo already spawned in {1}
INFO - [1/bar waiting job:00 flows:1] => waiting(queued)
CRITICAL - [1/foo failed job:01 flows:1] task proxy removed (suicide)
```

#### Additional context

(Update: I think this next bit might have been fixed by #4906 actually)

In some circumstances you might get `foo failed -> waiting` before the suicide message and then `foo` submits a second time. This might be why I had the mysterious long-lived "preparing" jobs in #4994. This is hard to reproduce ~(may be a random combination of pausing/resuming/holding/releasing/stopping/restarting that did it)~ (probably fixed by #4906).

```
INFO - [3/recover running job:01 flows:1] => succeeded
INFO - [3/foo failed job:01 flows:1] => waiting
INFO - [3/foo waiting job:01 flows:1] => waiting(queued)
CRITICAL - [3/foo waiting(queued) job:01 flows:1] task proxy removed (suicide)
...
INFO - [3/foo waiting job:01 flows:1] => preparing
...
INFO - [3/foo preparing job:02 flows:1] host=localhost
```

### PR DESCRIPTION
**Pull request by hjoliver**
These changes close #5002

I've run the example out to 50 cycle points several times, with nothing strange happening 

I haven't thought of a reliable way to reproduce the problem (which was more of an inefficiency + confusing warning than a balls-to-the-wall bug) in a short test. But codecov says the change is covered in a strictly technical sense at least.

**Requirements check-list**
- [x] I have read `CONTRIBUTING.md` and added my name as a Code Contributor.
- [x] Contains logically grouped changes (else tidy your branch by rebase).
- [x] Does not contain off-topic changes (use other PRs for other changes).
- [x] Applied any dependency changes to both `setup.cfg` and `conda-environment.yml`.
- [x] Already covered by existing tests.
- [x] Appropriate change log entry included.
- [x] No documentation update required.

### CODE REVIEW

**Comment by hjoliver** [comment]
Explanation:

Consider a task `foo` that depends only on an absolute trigger:
```
P1 = "init[3] => foo"
```

In principle, `init[3]` should spawn all `foo` instances, but that's not practically possible (there could be a large of infinite number of `foo` instances). So we have to autospawn `foo` instances as they were parentless, but make sure that none of them run before `init[3]` succeeds, or not spawn any `foo` until `init[3]` succeeds.

On current master, we spawn `3/foo` after `init:succeed`, and then autospawn other `foo` instances (out to the runahead limit) as if they are parentless. However, that's tricky and has an edge case problem.

On this branch, we autospawn all `foo` immediately (to the runahead limit), but they stay in the hidden pool (which holds partially satisfied prerequisites) with an unsatisfied prerequisite on `init:succeed`, until that gets satisfied.

This is simpler, and it works in all cases.

**Review by dwsutherland — APPROVED** [review]
Was able to reproduce:
```
(flow) sutherlander@graphic-vbox:~$ cylc cat-log parentless | grep WARNING
2022-07-25T15:00:07+12:00 WARNING - Backward compatibility mode ON
2022-07-25T15:00:14+12:00 WARNING - Task 3/foo already spawned in {1}
2022-07-25T15:00:14+12:00 WARNING - Task 5/foo already spawned in {1}
2022-07-25T15:00:14+12:00 WARNING - Task 5/foo already spawned in {1}
2022-07-25T15:00:57+12:00 WARNING - Task 23/foo already spawned in {1}
```
Fixed after this change:
```
(flow) sutherlander@graphic-vbox:~$ cylc cat-log parentless | grep WARNING
2022-07-25T15:02:59+12:00 WARNING - Backward compatibility mode ON
```

Code change looks fine

**Comment by wxtim** [comment]
Booted @MetRonnie as reviewer because I was able to reproduce issue locally.

**Review by wxtim — APPROVED** [review]
- [x] Read the code
- [x] Checked out and run tests locally
- [ ] Run Local Proof of pudding tests up to cycle 50

File: tests/integration/test_task_pool.py [hunk 1]
```diff
@@ -47,7 +49,7 @@
         'runahead limit': 'P3',
         'graph': {
             'P1': 'foo & bar',
-            'R1/2': 'foo[1] => pub'  # 2/pub doesn't spawn at start
+            'R1/2': 'foo[1] => pub'
```
Comment (MetRonnie) [inline tests/integration/test_task_pool.py:52]: Can we add a task to this test that doesn't get spawned to the runahead limit? (That was the point of `pub` originally, I think)

Comment (hjoliver) [reply tests/integration/test_task_pool.py:52]: Yes, I'd like to spend a bit of time making these integration tests more comprehensive.  As I quick fix, I just adapted the existing test to what's actually going on in the scheduler now.

Comment (hjoliver) [reply tests/integration/test_task_pool.py:52]: New issue: #5012

**Pull request merged**